Kudu.Services/Docker/DockerController.cs

```diff
@@ -50,5 +53,32 @@ public HttpResponseMessage ReceiveHook()
 
             return Request.CreateResponse(HttpStatusCode.OK);
         }
+
+        private static void FireSyncTriggers(IEnvironment environment, ITraceFactory traceFactory)
+        {
+            // create background tracer independent of request lifetime
+            var bgTracer = traceFactory.GetTracer();
+            bgTracer.Trace("DockerController.FireSyncTriggers");
+
+            // start new task to detach from request sync context
+            Task.Run(async () =>
+            {
+                try
+                {
+                    // When a docker image is updated, we need to ensure that a SyncTriggers operation is performed.
+                    // There are 2 cases:
+                    // 1) No containers are currently running - in this case the sync will start a container with the new image,
+                    //    and the sync will be up to date.
+                    // 2) One or more containers are currently running - in this case the container restart will cause them to
+                    //    restart, and the background sync triggers performed by the runtime will do the sync.
+                    // We do this on a background thread so we don't block the webhook on it. The sync is best effort.
+                    await FunctionsHelper.SyncTriggersAsync(bgTracer);
+                }
+                catch (Exception ex)
+                {
+                    bgTracer.TraceError(ex);
```

Comment:
Chose to do this in the background. Initially I thought it might be good to await the call as part of the webhook - that way if the sync fails, the caller might retry.

Reply:
The current syntrigger failed deployment if sync failed.  This would divert from that, no?

Reply:
Good point. I went back and forth on this... I'll switch to awaiting this